kolibri/content/utils/channel_import.py

```diff
@@ -0,0 +1,287 @@
+                # as not available in SQLite, only MySQL as far as I can tell:
+                # http://hackthology.com/how-to-compile-mysqls-on-duplicate-key-update-in-sql-alchemy.html
+                self.destination.session.merge(DestinationRecord(**data))
+            else:
+                data_to_insert.append(data)
+            unflushed_rows += 1
+            if unflushed_rows == 10000:
+                if not merge:
+                    self.destination.session.bulk_insert_mappings(DestinationRecord, data_to_insert)
+                    data_to_insert = []
+                self.destination.session.flush()
+                unflushed_rows = 0
+        if not merge and data_to_insert:
+            self.destination.session.bulk_insert_mappings(DestinationRecord, data_to_insert)
+        return unflushed_rows
+
+    def import_channel_data(self):
+
+        unflushed_rows = 0
+
+        for model in self.content_models:
+            mapping = self.schema_mapping.get(model, {})
+            row_mapper = self.generate_row_mapper(mapping.get('per_row'))
+            table_mapper = self.generate_table_mapper(mapping.get('per_table'))
+            logging.info('Importing {model} data'.format(model=model.__name__))
+            unflushed_rows = self.table_import(model, row_mapper, table_mapper, unflushed_rows)
+        self.destination.session.commit()
+
+    def delete_content_tree_and_files(self):
+        delete_content_tree_and_files(self.channel_id)
+
+    def end(self):
+        self.source.end()
+        self.destination.end()
+
+
+class NoVersionChannelImport(ChannelImport):
+    """
+    Class defining the schema mapping for importing old content databases (i.e. ones produced before the
+    ChannelImport machinery was implemented). The schema mapping below defines how to bring in information
+    from the old version of the Kolibri content databases into the database for the current version of Kolibri.
+    """
+
+    schema_mapping = {
+        ContentNode: {
+            'per_row': {
+                'channel_id': 'infer_channel_id_from_source',
+                'tree_id': 'get_tree_id',
+                'available': 'none',
+            },
+        },
+        File: {
+            'per_row': {
+                File._meta.get_field('local_file').attname: 'checksum',
+                'available': 'none',
+            },
+        },
+        LocalFile: {
+            'per_table': 'generate_local_file_from_file',
+            'per_row': {
+                'id': 'checksum',
+                'extension': 'extension',
+                'file_size': 'file_size',
+                'available': 'none',
+            },
+        },
+        ChannelMetadata: {
+            'per_row': {
+                ChannelMetadata._meta.get_field('min_kolibri_version').attname: 'set_version_to_no_version',
+                'root_id': 'root_pk',
+            },
+        },
+    }
```

Comment:
I guess if it were returned by a method, rather than being a hardcoded attribute, then you could call `super` and add/remove/modify things in the mapping, from the subclass?